src/services/localization/localization.js

```diff
@@ -0,0 +1,168 @@
+// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.
+
+// this file provides functionality for adding additonal languages to the project.
+
+/***********************************************************************************************
+imports
+***********************************************************************************************/
+
+import i18n from 'i18n-js';
+import memoize from 'lodash.memoize';
+import { I18nManager } from 'react-native';
+import * as RNLocalize from 'react-native-localize';
+import guestClient from '../rest/guestClient';
+
+import de from '../../CUSTOMIZATION/translations/de';
+import en from '../../CUSTOMIZATION/translations/en';
+import fr from '../../CUSTOMIZATION/translations/fr';
+import ar from '../../CUSTOMIZATION/translations/ar';
+
+/***********************************************************************************************
+constants
+***********************************************************************************************/
+
+// the fallback
+// this is the default the app will fallback to in case nothing else matches.
+// (for example: the device language is hindi, but only "de", "en", "fr", "ar" are provided by the app. So the last resort fallback would be "de")
+const defaultLanguage = 'de';
+
+// the fallback
+// this is the default reading direction the app will fallback to in case nothing else matches.
+const defaultIsRTL = false;
+
+// the available files / languages:
+// for each file available in in app/src/CUSTOMIZATION/translations an entry must be created here.
+// the attribute name shall be the language code the file represents.
+// if a language is present that is not provided with a matching questionnaire fom the backend, the language will remain available -
+// the user will then be presented with the default questionnaire language.
+const availableLanguageFiles = {
+  de: {
+    file: de,
+    isRTL: false,
+    title: 'Deutsch',
+  },
+
+  en: {
+    file: en,
+    isRTL: false,
+    title: 'English',
+  },
+
+  fr: {
+    file: fr,
+    isRTL: false,
+    title: 'Français',
+  },
+
+  ar: {
+    file: ar,
+    isRTL: true,
+    title: 'عربي',
+  },
+};
+
+// is used as the final composit of available languages
+const availableLanguages = {
+  ...availableLanguageFiles,
+};
+
+// function that memoizes the results of i18n.t. Also determines the cache key for storing the result
```

Comment:
typo: memorize

Reply:
I actually meant ["memoizes"](https://en.wiktionary.org/wiki/memoizes) 😅 .
But memorize is probably the better choice.